pkg/state/state.go

```diff
@@ -0,0 +1,104 @@
+package state
```

Comment:
What about moving this to internal? I understand that it could be a reusable one but it seems very tight to nigiri's state.